diff/postProcess.go

```diff
@@ -175,11 +175,18 @@ func (crud *targetPostAction) Create(arg ...crud.Arg) (crud.Arg, error) {
 	if !ok {
 		panic("whoops")
 	}
-	svc, ok := arg[1].(*state.Target)
+	target, ok := arg[1].(*state.Target)
 	if !ok {
 		panic("whoops")
 	}
-	s.Targets.Add(*svc)
+
+	u, err := s.Upstreams.Get(*target.Upstream.ID)
+	if err != nil {
+		return nil, err
+	}
+	target.Upstream = u.DeepCopy()
+
+	s.Targets.Add(*target)
```

Comment:
Could you explain the reason for doing these changes?

Other entities which have a foreign relation (like routes and services), don't actually search for a service and associate it with the route before adding/updating.

These shouldn't be required from what I can see. I might be missing something here.

Reply:
If we didn't set the upstream field here, the `Upstream` field of `target` here will have only a `ID` field. As we require `target.Upstream.Name` in `TargetsCollection `, this `s.Targets.Add(*target)` will return an error here.
And as the error is ignored here, this may cause problem for later usage of `s.Targets`.